packages/diagrams/lib/src/text.dart

```diff
@@ -484,6 +486,514 @@ class TextHeightComparisonPainter extends CustomPainter {
+    required bool shouldPaintCaptions,
+    required bool shouldPaintLeadingIndicator,
+  })  : _backgroundColor = backgroundColor,
+        _shouldPaintHeightIndicator = shouldPaintHeightIndicator,
+        _shouldPaintLeadingIndicator = shouldPaintLeadingIndicator,
+        _shouldPaintCaptions = shouldPaintCaptions,
+        textPainter = TextPainter(
+          textDirection: textDirection,
+          text: text,
+          textHeightBehavior: textHeightBehavior,
+        );
+
+  final TextPainter textPainter;
+
+  TextDirection get textDirection => textPainter.textDirection!;
+  set textDirection(TextDirection value) {
+    if (textPainter.textDirection == value)
+      return;
+    textPainter.textDirection = value;
+    markNeedsLayout();
+  }
+
+  TextSpan get text => textPainter.text as TextSpan;
+  set text(TextSpan? value) {
+    if (textPainter.text == value)
+      return;
+    textPainter.text = value;
+    markNeedsLayout();
+  }
+
+  Color get backgroundColor => _backgroundColor;
+  Color _backgroundColor;
+  set backgroundColor(Color value) {
+    if (value == _backgroundColor) {
+      return;
+    }
+    _backgroundColor = value;
+    markNeedsPaint();
+  }
+
+  TextHeightBehavior? get textHeightBehavior => textPainter.textHeightBehavior;
+  set textHeightBehavior(TextHeightBehavior? newValue) {
+    if (textHeightBehavior == newValue) {
+      return;
+    }
+    textPainter.textHeightBehavior = newValue;
+    markNeedsLayout();
+  }
+
+  bool get shouldPaintHeightIndicator => _shouldPaintHeightIndicator;
+  bool _shouldPaintHeightIndicator;
+  set shouldPaintHeightIndicator(bool value) {
+    if (value == _shouldPaintHeightIndicator)
+      return;
+
+    _shouldPaintHeightIndicator = value;
+    markNeedsPaint();
+  }
+
+  bool get shouldPaintLeadingIndicator => _shouldPaintLeadingIndicator;
+  bool _shouldPaintLeadingIndicator;
+  set shouldPaintLeadingIndicator(bool value) {
+    if (value == _shouldPaintLeadingIndicator)
+      return;
+    _shouldPaintLeadingIndicator = value;
+    markNeedsPaint();
+  }
+
+  bool get shouldPaintCaptions => _shouldPaintCaptions;
+  bool _shouldPaintCaptions;
+  set shouldPaintCaptions(bool value) {
+    if (value == _shouldPaintCaptions)
+      return;
+    _shouldPaintCaptions = value;
+    markNeedsPaint();
+  }
+
+  late final TextPainter heightCaptionTextPainter = TextPainter(textDirection: textDirection);
+  static late final TextStyle heightCaptionTextStyle = TextStyle(fontSize: 20, color: Colors.blue[900]!);
+  static const Offset fontMetricsLabelPadding = Offset(0, 3);
+  static const double heightCaptionBracketMinX = -25;
+
+  @override
+  Size computeDryLayout(BoxConstraints constraints) {
+    textPainter.layout(minWidth: constraints.minWidth, maxWidth: double.infinity);
+    if (shouldPaintHeightIndicator) {
+      heightCaptionTextPainter.text = TextSpan(text: 'line-height:\n${textPainter.height} px', style: heightCaptionTextStyle);
```

Comment:
The use of hyphens here for line-height and line-box makes me think that this is referring to something specific in Flutter (or it makes me think of CSS).  It might be confusing to users when they see this diagram show that line-height is 360px, but there is no direct line-height or lineHeight parameter on EditableText that can be used to set it to 360px.  In the docs that use this diagram, we should be clear about where these values come from and how they can be achieved in Flutter.

Reply:
Ah good point. How do "Text Height"  and "Text Top/Bottom" sound?